open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,608 @@
+
+            }
+
+        } else {
+            // If periodic limits key is missing, return an error
+            log.error(ErrorConstants.MISSING_PERIOD_LIMITS);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.INVALID_REQ_PAYLOAD,
+                    ErrorConstants.MISSING_PERIOD_LIMITS,
+                    ErrorConstants.PATH_PERIOD_TYPE);
+        }
+
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+
+    /**
+     * Validates the date-time parameters in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response. If the date-time parameters are valid,
+     * it sets the "IS_VALID" field to true; otherwise, it contains an error response.
+     */
+    public static JSONObject validateParameterDateTime(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        if (controlParameters.containsKey(ConsentExtensionConstants.VALID_TO_DATE_TIME)) {
+
+            // Retrieve the validDateTime from the control parameters
+            Object validateToDateTime = controlParameters.get(ConsentExtensionConstants.VALID_TO_DATE_TIME);
+
+            if (!isValidDateTimeObject(validateToDateTime)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_VALID_TO_DATE,
+                        ErrorConstants.INVALID_DATE_TIME_FORMAT,
+                        ErrorConstants.PATH_VALID_TO_DATE);
+            }
+        }
+
+        if (controlParameters.containsKey(ConsentExtensionConstants.VALID_FROM_DATE_TIME)) {
+
+            // Retrieve the validDateTime from the control parameters
+            Object validateFromDateTime = controlParameters.get(ConsentExtensionConstants.VALID_FROM_DATE_TIME);
+
+            if (!isValidDateTimeObject(validateFromDateTime)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_VALID_FROM_DATE,
+                        ErrorConstants.INVALID_DATE_TIME_FORMAT,
+                        ErrorConstants.PATH_VALID_TO_DATE);
+            }
+        }
+
+        OffsetDateTime validToDateTime = OffsetDateTime.parse(controlParameters.
+                getAsString(ConsentExtensionConstants.VALID_TO_DATE_TIME));
+        OffsetDateTime validFromDateTime = OffsetDateTime.parse(controlParameters.
+                getAsString(ConsentExtensionConstants.VALID_FROM_DATE_TIME));
+        OffsetDateTime currentDateTime = OffsetDateTime.now(validToDateTime.getOffset());
+
+        //If the ValidToDAte is older than current date OR currentDate is older than ValidFromDAte, return error
+        if (!validFromDateTime.isBefore(currentDateTime) || !currentDateTime.isBefore(validToDateTime)) {
+            log.debug("Invalid date-time range.");
```

Comment:
Shall we add more meaningful debug logs here with the values as well,

Ex: `log.error(String.format("Invalid date-time range, validToDateTime : %s , validFromDateTime : %s and currentDateTime : %s ", validToDateTime, validFromDateTime,  currentDateTime));`

Reply:
Fixed in [5ba1c85]